test/group0_tables/test_group0_tables.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import enum
+
+
+def random_string(size=10) -> str:
+    """Generate a random string of fixed length """
+    letters = string.ascii_lowercase
+    return ''.join(random.choice(letters) for _ in range(size))
+
+
+async def select(cql, key: str) -> Optional[str]:
+    result = list(await cql.run_async(f"SELECT value FROM system.group0_kv_store WHERE key = '{key}';"))
+
+    if len(result) == 0:
+        return None
+
+    assert len(result) == 1
+    return result[0].value
+
+async def update(cql, key: str, new_value: str) -> None:
+    await cql.run_async(f"UPDATE system.group0_kv_store SET value = '{new_value}' WHERE key = '{key}';")
+
+async def update_conditional(cql, key: str, new_value: str, value_condition: Optional[str]) -> Tuple[bool, Optional[str]]:
+    condition = "null" if value_condition is None else f"'{value_condition}'"
+    result = await cql.run_async(f"UPDATE system.group0_kv_store SET value = '{new_value}' WHERE key = '{key}' IF value = {condition};")
+
+    assert len(result) == 1
+    return (result[0].applied, result[0].value)
+
+
+class QueryType(enum.Enum):
+    SELECT = enum.auto()
+    UPDATE = enum.auto()
+    CONDITIONAL_UPDATE = enum.auto()
+
+class ConditionType(enum.Enum):
+    RANDOM = enum.auto()
+    FULLFILLED = enum.auto()
+    NEW_KEY = enum.auto()
+
+@pytest.mark.asyncio
+async def test_group0_kv_store(cql) -> None:
+    experimental_features = list(await cql.run_async("SELECT value FROM system.config WHERE name = 'experimental_features'"))[0].value
+    assert "group0-tables" in experimental_features
+
+    random_strings: List[str] = [random_string() for _ in range(42)]
+    kv_store: Dict[str, str] = {}
+
+    query_types = random.choices(
+            [QueryType.SELECT, QueryType.UPDATE, QueryType.CONDITIONAL_UPDATE],
+            weights=[0.3, 0.2, 0.5], k=3721
+        )
+    for query_type in query_types:
+        key = random.choice(random_strings)
+
+        if query_type == QueryType.SELECT:
+            value = await select(cql, key)
+            assert value == kv_store.get(key)
+        else:
+            new_value = random.choice(random_strings)
+
+            if query_type == QueryType.UPDATE:
+                await update(cql, key, new_value)
+                kv_store[key] = new_value
+            elif query_type == QueryType.CONDITIONAL_UPDATE:
+                value_condition: Optional[str]
+                condition_type = random.choice(list(ConditionType))
+                if condition_type == ConditionType.RANDOM:
+                    value_condition = random.choice(list[Optional[str]](random_strings) + [None])
+                elif condition_type == ConditionType.FULLFILLED:
+                    value_condition = kv_store.get(key)
+                elif condition_type == ConditionType.NEW_KEY:
+                    key = random_string()
+                    random_strings.append(key)
+                    value_condition = None
```

Comment:
I'm worried that the probability of choosing a key that already exists in `kv_store` is quite low with this approach.
Each of these cases gives a high chance that we'll choose a new key (where by new I mean not present in `kv_store`)

I would change the set of ConditionTypes to something like `[NEW_KEY, EXISTING_KEY_PASS, EXISTING_KEY_FAIL]` with about 20% chance for `NEW_KEY` (unless of course `kv_store` is empty, then choose `NEW_KEY`). For `NEW_KEY` we would randomly choose from the set difference `random_strings - keys_in_kv_store` (I don't think it's necessary to generate any more random strings, 42 is enough), for `EXISTING_KEY_*` we would randomly choose from `keys_in_kv_store` (and if `keys_in_kv_store` covers all of `random_strings`, we would always choose `EXISTING_KEY_*`). Then `PASS` means we pass the condition (so `value_condition` takes `kv_store[key]`), `FAIL` means we don't pass (so `value_condition` takes something different from `kv_store[key]`). `FAIL` vs `PASS` chance could be 50-50.

Also, the test should be reproducible if possible. We could roll the random generator seed at the beginning, print it, then pass it to the random generator. If one wants to reproduce the sequence of operations, they can reuse this seed.